src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -839,11 +854,40 @@ public void SetAbi (string abi)
 			int code = 1;
 			if (!string.IsNullOrEmpty (VersionCode)) {
 				code = Convert.ToInt32 (VersionCode);
-				if (code > 0xffff || code < 0)
-					throw new ArgumentOutOfRangeException ("VersionCode", "VersionCode is outside 0, 65535 interval");
+				if (code > maxVersionCode || code < 0)
+					throw new ArgumentOutOfRangeException ("VersionCode", $"VersionCode is outside 0, {maxVersionCode} interval");
 			}
 			code |= GetAbiCode (abi) << 16;
 			VersionCode = code.ToString ();
 		}
+
+		public void CalculateVersionCode (string currentAbi, string versionCodePattern, string versionCodeProperties)
+		{
+			var regex = new Regex ("\\{(?<key>([A-Za-z]*)):?0*[\\}]");
+			var kvp = new Dictionary<string, int> ();
+			foreach (var item in versionCodeProperties?.Split (new char [] { ';' }) ?? new string [0]) {
```

Comment:
This should split on both `;` and `:`, or just `:`.

`;` can't be specified on the command-line as part of a property value, as it's the property separator:

    msbuild /p:Foo=Bar;Baz=Qux